### PR TITLE
CSHARP-3355: Add more spec tests for invalid relaxed UUIDs.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**
EG: https://evergreen.mongodb.com/version/601d956b3e8e86586c978855

### CODE REVIEW

File: tests/MongoDB.Bson.Tests/Specifications/bson-corpus/tests/bsonview [hunk 1]
```diff
@@ -0,0 +1,434 @@
+#!/usr/bin/env perl
```
Comment (jyemin) [inline tests/MongoDB.Bson.Tests/Specifications/bson-corpus/tests/bsonview:1]: I think this file can be deleted

**Review by jyemin — CHANGES_REQUESTED** [review]

File: tests/MongoDB.Bson.Tests/Specifications/bson-corpus/tests/bsonview (hunk 1 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Bson.Tests/Specifications/bson-corpus/tests/bsonview:1]: as far as I know, we copy such files under git to make a diff with the spec folder easier. So, if we have such a file under git we can just fully copy all files from the spec folder into our tests without any analysis

File: src/MongoDB.Bson/IO/JsonReader.cs [hunk 2]
```diff
@@ -2089,14 +2090,25 @@ private BsonValue ParseUuidExtendedJson()
...
 
+            if (uuidTokenStringValue.Count(c => c == '-') != 4 ||
+                uuidTokenStringValue[8] != '-' ||
+                uuidTokenStringValue[13] != '-' ||
+                uuidTokenStringValue[18] != '-' ||
+                uuidTokenStringValue[23] != '-')
+            {
+                throw new FormatException($"Invalid $uuid hyphens format: '{uuidTokenStringValue}'.");
+            }
+
```
Comment (rstam) [inline src/MongoDB.Bson/IO/JsonReader.cs:2111]: I think you can replace lines 2093-2112 with:

    var guid = Guid.Parse(uuidToken.StringValue);
    var binaryData = new BsonBinaryData(guid, GuidRepresentation.Standard);

`Guid.Parse` already does all the validations we need.

File: tests/MongoDB.Bson.Tests/Specifications/bson-corpus/tests/bsonview (hunk 1 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Bson.Tests/Specifications/bson-corpus/tests/bsonview:1]: at least mostly :)

**Review by rstam — CHANGES_REQUESTED** [review]

**Comment by DmitryLukyanov** [comment]
EG: https://evergreen.mongodb.com/version/601d9af00305b9115d9a8acd

File: src/MongoDB.Bson/IO/JsonReader.cs (hunk 2 above)
Comment (DmitryLukyanov) [reply src/MongoDB.Bson/IO/JsonReader.cs]: good point, done

File: tests/MongoDB.Bson.Tests/Specifications/bson-corpus/tests/bsonview (hunk 1 above)
Comment (jyemin) [reply tests/MongoDB.Bson.Tests/Specifications/bson-corpus/tests/bsonview:1]: If that's the policy, then so be it, but this is a weird one to copy.

**Comment by jyemin** [comment]
Removed James since Robert picked it up.

File: tests/MongoDB.Bson.Tests/Specifications/bson-corpus/tests/bsonview (hunk 1 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Bson.Tests/Specifications/bson-corpus/tests/bsonview:1]: it's not a defined policy, but this is how we did previously

**Review by rstam — APPROVED** [review]
LGTM

File: src/MongoDB.Bson/IO/JsonReader.cs [hunk 3]
```diff
@@ -18,6 +18,7 @@
 using System.Globalization;
 using System.IO;
 using System.Text.RegularExpressions;
+using System.Linq;
```
Comment (JamesKovacs) [inline src/MongoDB.Bson/IO/JsonReader.cs:21]: Not used.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Changes look good. Verified that:
- BsonBinarySubType is still set correctly.
- Same exception type (FormatException) is thrown as before.
- Tests run successfully.

One minor change required. You have an extraneous `using` statement.

File: src/MongoDB.Bson/IO/JsonReader.cs (hunk 3 above)
Comment (DmitryLukyanov) [reply src/MongoDB.Bson/IO/JsonReader.cs]: Done

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**